Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp

```diff
@@ -88,48 +88,46 @@ struct TestTrackState {
+  //
+  // @param[in] pc TestTrackState with the input values
+  // @param[in] mask Specifies which components are used/filled
+  // @param[out] ts TrackStateProxy which is filled
+  // @param [in] measdim Dimension of the measurement
+  template <typename trajectory_t, typename track_state_t>
+  void fillTrackState(TrackStatePropMask mask,
+                      track_state_t& ts) {
+    // always set the reference surface
+    ts.setReferenceSurface(predicted.referenceSurface().getSharedPtr());
 
-  if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Predicted)) {
-    ts.predicted() = pc.predicted.parameters();
-    assert(pc.predicted.covariance().has_value());
-    ts.predictedCovariance() = *(pc.predicted.covariance());
-  }
-  if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Filtered)) {
-    ts.filtered() = pc.filtered.parameters();
-    assert(pc.filtered.covariance().has_value());
-    ts.filteredCovariance() = *(pc.filtered.covariance());
-  }
-  if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Smoothed)) {
-    ts.smoothed() = pc.smoothed.parameters();
-    assert(pc.smoothed.covariance().has_value());
-    ts.smoothedCovariance() = *(pc.smoothed.covariance());
-  }
-  if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Jacobian)) {
-    ts.jacobian() = pc.jacobian;
-  }
-  ts.chi2() = pc.chi2;
-  ts.pathLength() = pc.pathLength;
-  // source link defines the uncalibrated measurement
-  // create calibrated measurements from source link
-  if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Calibrated)) {
-    testSourceLinkCalibrator<trajectory_t>(Acts::GeometryContext{},
-                                           Acts::CalibrationContext{},
-                                           SourceLink{pc.sourceLink}, ts);
-    assert(ts.hasUncalibratedSourceLink());
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Predicted)) {
+      ts.predicted() = predicted.parameters();
+      assert(predicted.covariance().has_value());
+      ts.predictedCovariance() = *(predicted.covariance());
+    }
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Filtered)) {
+      ts.filtered() = filtered.parameters();
+      assert(filtered.covariance().has_value());
+      ts.filteredCovariance() = *(filtered.covariance());
+    }
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Smoothed)) {
+      ts.smoothed() = smoothed.parameters();
+      assert(smoothed.covariance().has_value());
+      ts.smoothedCovariance() = *(smoothed.covariance());
+    }
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Jacobian)) {
+      ts.jacobian() = jacobian;
+    }
+    ts.chi2() = chi2;
+    ts.pathLength() = pathLength;
+    // source link defines the uncalibrated measurement
+    ts.setUncalibratedSourceLink(Acts::SourceLink{sourceLink});
+    // create calibrated measurements from source link
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Calibrated)) {
+      testSourceLinkCalibrator<trajectory_t>(Acts::GeometryContext{}, ts);
```

Comment:
The calibration context goes away?

Reply:
I just made fillTrackState a method of TestTrackState. 
The sourceLink related code is still not compiling though. 
Wanted to hear about if the direction is right one.

Reply:
I understand now. This is an effect of git trying to resolve the conflict. Adding back all needed args.